### PR TITLE
Adding dual pane support to PostListFragment 

### PR DESCRIPTION
**Pull request by khaykov**
This is the first fragment to be used in dual pane mode.

Fragments used in dual pane layout require some special logic - toolbar should be inside fragment, visible in single pane, and hidden in dual pane, content margins should also be different for both modes.

PostListFragment was almost ready to use as-is, I just moved toolbar from activity to fragment, added UI logic for dual pane mode, added newInstance() so the fragment could be initialized programmatically, and removed setRetainInstance(true) which was not really needed.

I also added DualPaneFragment superclass that would contain shared logic of fragments with dual pane support.

PR is the part of issue #3551

### CODE REVIEW

File: WordPress/src/main/res/layout/post_list_fragment.xml [hunk 1]
```diff
@@ -1,85 +1,95 @@
...
-
-<RelativeLayout xmlns:android="http://schemas.android.com/apk/res/android"
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
     xmlns:app="http://schemas.android.com/apk/res-auto"
     xmlns:tools="http://schemas.android.com/tools"
-    android:id="@+id/root_view"
     android:layout_width="match_parent"
-    android:layout_height="match_parent">
+    android:layout_height="match_parent"
+    android:background="?android:attr/windowBackground"
```
Comment (maxme) [inline WordPress/src/main/res/layout/post_list_fragment.xml]: This is going to add a lot of overdraw (fullscreen overdraw actually).

Comment (khaykov) [inline WordPress/src/main/res/layout/post_list_fragment.xml]: It will! Thank you for pointing it out, I totally missed the overdraw issue!

**Comment by khaykov** [comment]
Based on @maxme suggestion I reduced overdraw and layout nesting. I also removed background from empty_view (it was drawing same background as window). I think I figured out how achieve desired functionality without adding extra background layer.

**Comment by maxme** [comment]
Looks better with the fix! :shipit:

**Pull request merged**